app/forms/bulk_submission_form.rb

```diff
@@ -78,4 +92,96 @@ def file_size(file)
     return 0 if file.nil?
     File.size(file&.tempfile)
   end
+
+  def csv
+    @csv ||= CSV.parse(File.read(uploaded_file), headers: true)
+  rescue StandardError
+    errors.add(:uploaded_file, :unparseable_file, filename: uploaded_file.original_filename)
+    nil
```

Comment:
do we need the nil. the fact that have not raised the error should suffice

Reply:
I do still need this test unfortunately, so that it doesn't attempt to validate_file_contents if the csv is unparseable. The errors.empty? check will only work if the error is raised during the initial validations (file_chosen, file_empty etc.)